src/CLR/Core/CLR_RT_HeapBlock.cpp

```
@@ -867,10 +879,31 @@ bool CLR_RT_HeapBlock::TypeDescriptorsMatch(
 
         case DATATYPE_SZARRAY:
         {
-            // compare outer dims (always 1) then element types
-            CLR_RT_TypeDescriptor expectedElementType, actualElementType;
-            expectedElementType.GetElementType(expectedElementType);
-            actualElementType.GetElementType(actualElementType);
+            // derive element-type descriptors from the ARRAY descriptors
+            CLR_RT_TypeDescriptor expectedElementType;
```

Clang (14.0.6)

[warning] uninitialized record type: 'expectedElementType'

(cppcoreguidelines-pro-type-member-init)